nx-X11/programs/Xserver/os/utils.c

```diff
@@ -556,7 +556,12 @@ AdjustWaitForDelay (void * waitTime, unsigned long newdelay)
 
 void UseMsg(void)
 {
+#ifdef NXAGENT_SERVER
+    extern const char *__progname;
+    ErrorF("Usage: %s [<options>] [:<display>]\n\n", __progname);
```

Comment:
We should avoid `__progname`, since it's not portable. Can you instead pass `argv[0]` around?

Reply:
> We should avoid __progname, since it's not portable. Can you instead pass argv[0] around?

I left the original commit as is and added another commit that drops __progname usage from the entire nxagent (we already used it elsewhere). The commit is also in this PR.